apps/src/util/dataStructures/WeakMapPlus.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * A small improvement to JavaScript/TypeScript's builtin `WeakMap` that adds support
+ * for `undefined` as a key.  `WeakMap` does not support primitive keys other than
+ * Symbol (and this only in later specs).
+ **/
+export class WeakMapPlus extends WeakMap {
+  /*
+   * WeakMap compatiple key to swap for `undefined`.  Currently uses an object
+   * but could use a symbol if our minimum supported browsers implement the spec
+   * or if our transpilers (babel/tsc) can add a polyfill implementation .
+   * See: https://github.com/tc39/ecma262/pull/2777
+   **/
+  static {
+    this._undefinedKey = {};
+  }
```

Comment:
Is there a specific reason to use an initialization block here, rather than just

```suggestion
  static _undefinedKey = {};
```

?

Reply:
Nope!  I ran the original TypeScript through the compiler with `ESNext` as the target (compiling it to ES5 was breaking things) and didn't notice it did that.